modules/47_regipol/regiCarbonPrice/datainput.gms

```diff
@@ -18,9 +18,17 @@ p47_implicitQttyTargetTaxRescale_iter("1", "2030",ext_regi,qttyTarget,qttyTarget
 $endIf.cm_implicitQttyTarget
 
 *** RR this should be replaced as soon as non-energy is treated endogenously in the model
-*** EUR in 2030 ~ 90Mtoe (90 * 10^6 toe -> 90 * 10^6 toe * 41.868 GJ/toe -> 3768.12 * 10^6 GJ * 10^-9 EJ/GJ -> 3.76812 EJ * 1 TWa/31.536 EJ -> 0.1194863 TWa) EU27 = 92% EU28"
-p47_nonEnergyUse("2030",ext_regi)$(sameas(ext_regi, "EUR_regi")) = 0.1194863;
-p47_nonEnergyUse("2030",ext_regi)$(sameas(ext_regi, "EU27_regi")) = 0.11;
+*** https://ec.europa.eu/eurostat/databrowser/bookmark/f7c8aa0e-3cf6-45d6-b85c-f2e76e90b4aa?lang=en
+*** EU27 -> (2018:2021)/4 = 3.835 EJ = 0.121606 Twa (2020 non-energy use)
+*** EU27 -> 2030 non energy use is 91.6% of 2020 in pm_fe_nechem.cs4r (92.4% for EU28)
+p47_nonEnergyUse("2030",ext_regi)$(sameas(ext_regi, "EU27_regi")) = 0.121606*0.916; 
+p47_nonEnergyUse("2030",ext_regi)$(sameas(ext_regi, "EUR_regi")) = 0.13*0.924;
+p47_nonEnergyUse("2050",ext_regi)$(sameas(ext_regi, "EU27_regi")) = 0.121606*0.815; 
```

Comment:
where does the 0.815 come from?

Reply:
0.815 = 
   2050 total non-energy value in the exogenous non-energy trajectory defined at pm_fe_nechem.cs4r 
   /
   2020 total non-energy value in the exogenous non-energy trajectory defined at pm_fe_nechem.cs4r 

I wouldn't worry much about values, other than 2030 ones, as we don't currently use them in any target anyway.

Reply:
right, my point was not so much "are the exact values right", but rather on adding a short comment like "!! Values that give the reduction over time were estimated from previous REMIND runs. Should be updated once better information becomes available" or so, to clearly communicate to any future person looking at this that they should feel very free to change the numbers.

Reply:
There is already a comment saying where the numbers for EU27 and EU28 come from, above:
```
*** EU27 -> (2018:2021)/4 = 3.835 EJ = 0.121606 Twa (2020 non-energy use)
*** EU27 -> 2030 non energy use is 91.6% of 2020 in pm_fe_nechem.cs4r (92.4% for EU28)
```
and another one, just above, saying when these should be removed from the code in favor of the next implementation that will hopefully make those out of date:
```
*** RR this should be replaced as soon as non-energy is treated endogenously in the model
```
In the `pm_fe_nechem` code there are also comments already in place showing where that data came from, and how it was calculated.

I don't see the need to repeat this information all over again. Whoever needs more detail about this, can always simply click on the line in git, and see the last commit and PR related with these change to find this information here.

Reply:
https://github.com/remindmodel/remind/pull/1331/commits/d006d5410a7905563fc7f76dab95edbddb13badb